src/Datasource/QueryCacher.php

```diff
@@ -72,7 +65,7 @@ public function __construct($key, $config)
      * @param object $query The query the cache read is for.
      * @return mixed|null Either the cached results or null.
      */
-    public function fetch(object $query)
+    public function fetch(object $query): mixed
```

Comment:
```suggestion
    public function fetch(QueryInterface $query): ?ResultSetInterface
```

Similarly the signature for `store()` can also be changed to 
```
store(QueryInterface $query, ResultSetInterface $results): bool
```

and `resolveKey(object $query): string` to
```
resolveKey(QueryInterface $query): string
```

Reply:
I would prefer that to be a separate PR.

Reply:
Okay :+1: .